### ISSUE TITLE
Not implemented constructs

### ISSUE THREAD

**Issue by jayvdb** [adsharma/py2many#291]
We've talked a bit about how to handle not-implemented Python constructs.

Raising NotImplementedError, and building specialised more meaningful subclasses, is the road we're currently going down.

The problem with that approach is that it halts the transpiler at the first unsupported construct, preventing someone from seeing that most of their python input is able to be transpiled, and only a section of it is not supported.

Instead of this, we might enclose an un-transpiled segment into an assertion in the target language which causes the problem to fail when that part of the code in the target language is used.  The assertion would show the (ast of the) original Python code which couldnt be transpiled.  Ideally it also contains some constant in the generated code so that the transpiler invoker (e.g. the CLI, or test rig) can detect and report to the caller how many un-transpiled segments exist in the generated code.  This will also increase the coverage, because the transpiler will keep working after the unsupported node.

The transpilers currently have `visit_unsupported_body` which looks like it might have been trying to do something like this.

Ideally both modes are possible, as a py2many developer usually wants to see a backtrace, but an end-user wants a best-effort output.

**Comment by adsharma** [comment]
Yes - behavior could be controlled with a flag. The end user might also want correctness guarantees in the form of validation errors from the transpiler.
